internal/database/mysql.go

```diff
@@ -120,59 +118,73 @@ func (db *MySQLDatabase) GetUser(id int) (user *internal.User) {
 	return
 }
 
-func (db *MySQLDatabase) getOpenID() (id int) {
-	err := db.underlyingDB.QueryRow("SELECT id FROM openid LIMIT 1").Scan(&id)
+func (db *MySQLDatabase) getOpenID() (id uint64, err error) {
+	err = db.underlyingDB.QueryRow("SELECT id FROM openid LIMIT 1").Scan(&id)
 	if err != nil && !errors.Is(err, sql.ErrNoRows) {
 		log.Println("Failed to query for an openid -", err)
+		return 0, errors.New("invalid openid")
 	}
-	return
+	return id, nil
 }
 
-func (db *MySQLDatabase) AddUser(username string, email string, password string) (err error) {
-
-	// TODO: There needs to be a check if the usernames/emails/passwords empty here.
-	if len(username) == 0 || len(email) == 0 || len(password) == 0 {
-		return errors.New("empty fields")
-	}
-	// TODO(@seoyoungcho213): For efficiency, we might be able to return the new id here with only a single query?``
-	query := "INSERT INTO users (username, email, password"
-	if openID := db.getOpenID(); openID == 0 {
-		log.Println("All existing IDs in use, assigning new ID to {" + username + "}")
-		query += `) VALUES ("%s", "%s", "%s")`
+func (db *MySQLDatabase) AddUser(first_name, middle_name, last_name, email, password, date_of_birth string) (
+	id uint64, err error) {
+	if len(first_name) == 0 || len(last_name) == 0 || len(email) == 0 || len(password) == 0 || len(date_of_birth) == 0 {
+		return 0, errors.New("empty fields")
+	}
+	query := "INSERT INTO users (first_name, middle_name, last_name, email, password, date_of_birth"
+	if openID, _ := db.getOpenID(); openID == 0 {
```

Comment:
we need to have a really good reason not to be handling the error here.

Reply:
You can prolly do:


if openID, err := db.getOpenID(); err != nil {
} else if openID == 0 {
} else {
...
}
...